src/lib.rs

```diff
@@ -727,15 +727,219 @@ impl CertificateParams {
 		let alg = SignatureAlgorithm::from_oid(&alg_oid.collect::<Vec<_>>())?;
 
 		let dn = DistinguishedName::from_name(&x509.tbs_certificate.subject)?;
+		let is_ca = Self::convert_x509_is_ca(&x509)?;
+		let validity = x509.validity();
+		let serial = Self::convert_x509_serial(&x509);
+		let subject_alt_names = Self::convert_x509_subject_alternative_name(&x509)?;
+		let key_usages = Self::convert_x509_key_usages(&x509)?;
+		let extended_key_usages = Self::convert_x509_extended_key_usages(&x509)?;
+		let name_constraints = Self::convert_x509_name_constraints(&x509)?;
+
 		Ok(
 			CertificateParams {
 				alg,
+				is_ca,
+				subject_alt_names,
+				key_usages,
+				extended_key_usages,
+				name_constraints,
 				distinguished_name : dn,
 				key_pair : Some(key_pair),
+				serial_number: Some(serial),
+				not_before : validity.not_before.to_datetime(),
+				not_after : validity.not_after.to_datetime(),
 				.. Default::default()
 			}
 		)
 	}
+	#[cfg(feature = "x509-parser")]
+	fn convert_x509_is_ca(x509 :&x509_parser::certificate::X509Certificate<'_>) -> Result<IsCa, RcgenError> {
+		use x509_parser::extensions::BasicConstraints as B;
+
+		let basic_constraints = x509.basic_constraints()
+			.or(Err(RcgenError::CouldNotParseCertificate))?.map(|ext| ext.value);
+
+		let is_ca = match basic_constraints {
+			Some(B { ca: true, path_len_constraint: Some(n) }) => IsCa::Ca(BasicConstraints::Constrained(*n as u8)),
```

Comment:
This should return some error if n does not fit an u8 I think.

Reply:
Addressed in c1b0b40. I was unsure if just returning `Unconstrained` instead of an error would be better, but went with the error route.